### PR TITLE
fix(ansible-galaxy): Extend Ansible URL parser

### PR DESCRIPTION
**Pull request by timmjd**
## Changes:

Adapt the parsing capability for git based ansible galaxy source URLs.

## Context:

Current implementation did not cover anonymous git protocol to be chosen for clone operation and did not handle the case for `.` inside the repository name.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added unit tests, or
- [ ] No new tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/manager/ansible-galaxy/__snapshots__/extract.spec.ts.snap [hunk 1]
```diff
@@ -75,9 +75,21 @@ Array [
   },
   Object {
     "currentValue": "0.14",
-    "datasource": "galaxy",
-    "depName": "company.com:mygroup/ansible-base.git",
+    "datasource": "git-tags",
+    "depName": "mygroup/ansible-base",
```
Comment (timmjd) [inline lib/manager/ansible-galaxy/__snapshots__/extract.spec.ts.snap:79]: Should not be hit by the `galaxyRoleRegex` ( [namespace.role_name](https://docs.ansible.com/ansible/latest/galaxy/user_guide.html) ).

**Comment by timmjd** [comment]
Also related to the change #9387

**Comment by rarkins** [comment]
@secustor could you review?

**Comment by secustor** [comment]
@rarkins  LGTM

**Review by viceice — APPROVED** [review]

**Pull request merged**